util/dht.py

```diff
@@ -0,0 +1,42 @@
+import sensors.dht as dht
+import functools
+import time
+import asyncio
+import typing
+import config
+
+
+async def start_runner(config: config.SensorConfig):
+    reader = _get_appropriate_reader(config)
+    await _runner(reader)
+
+
+def print_reading(reading: dht.DHTReading):
+    t = time.localtime()
+    print("="*20)
+    print(f"Timestamp: {time.strftime('%H:%M:%S', t)}")
+    print(f"Code: {reading.code}")
+    print(f"Humidity: {reading.humidity}%")
+    print(f"Temperature: {reading.temperature}°C")
+
+
+ReaderCallback = typing.Callable[[], dht.DHTReading]
+
+
+async def _runner(reader: ReaderCallback):
+    while True:
+        reading = reader()
+        print_reading(reading)
+        await asyncio.sleep(2)
+
+
+def _make_reader(pin: int):
+    return functools.partial(dht.read_dht, pin=pin)
+
```

Comment:
Is this necessary? I'd rather have `_get_appropriate_reader` not be a single-line function than to have 2 private funcions.

Reply:
https://github.com/magley/my-pi-home/pull/1/commits/686e2564ed4392cb1e6f43fd872b50a305c828f2